### PR TITLE
Move autologin

### PR DESCRIPTION
**Pull request by SRFU-NN**
* Added logout
* Allow to run as context handler, with auto login and logout
* Auto logout on deletion of EmpowerConnection

### CODE REVIEW

File: tests/test_core_api.py [hunk 1]
```diff
@@ -6,31 +6,30 @@
...
-        # The connection logs in automatically, so we need to mock that response
+        # Since we log in, we need to mock that connection.
 
         mock_password = MagicMock()
         mock_password.return_value = "test_password"
         self.mock_password = mock_password
-        mock_getpass.return_value = mock_password
         # getpass is used to get the password, so we need to mock that response since
         # interactivity is not possible
         self.connection = EmpowerConnection(
```
Comment (SRFU-NN) [inline tests/test_core_api.py:24]: Test comment

File: src/OptiHPLCHandler/empower_api_core.py [hunk 2]
```diff
@@ -97,8 +100,25 @@ def login(self, password: Optional[str] = None) -> None:
             logger.error("Login failed")
             raise IOError("Login failed")
         self.token = reply.json()["results"][0]["token"]
+        self.session_id = reply.json()["results"][0]["id"]
         logger.debug("Login successful, keeping token")
 
+    def logout(self) -> None:
```
[the remark this answers is not in the document]
Comment (SRFU-NN) [reply src/OptiHPLCHandler/empower_api_core.py:98]: Implemented

File: src/OptiHPLCHandler/empower_api_core.py [hunk 3]
```diff
@@ -54,11 +52,8 @@ def __init__(
             time.
         """
         self.address = address.rstrip("/")  # Remove trailing slash if present
-        if username is None:
-            logger.debug("No username specified, getting username from system")
-            self.username = getpass.getuser()
-        else:
-            self.username = username
+        logger.debug("No username specified, getting username from system")
```
[the remark this answers is not in the document]
Comment (SRFU-NN) [reply src/OptiHPLCHandler/empower_api_core.py]: Implemented

**Review by Firefly78 — COMMENTED** [review]
Looks good.

**Pull request merged**